### PR TITLE
Upgrade libcanard to 3.0 and rename uavcan_v1 to cyphal

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Updated libcanard to 3.0 which did change the API a bit.

Renamed uavcan_v1 to cyphal
@pavel-kirienko 

Updated public_regulated_data_types repo

Introduced CanardHandle class for handling libcanard instance and tx buffer. Furthermore this decouples the physical interface from Cyphal.cpp which should make it easy to implement a redundant CAN class, which would be a future step.

### CODE REVIEW

**Comment by dagar** [comment]
OpenCyphal or just Cyphal?

**Comment by pavel-kirienko** [comment]
@dagar OpenCyphal is the project, Cyphal is the protocol. OpenCyphal aims to provide open implementations of Cyphal, but we anticipate third-party implementers to appear, too, such as https://github.com/samcrow/canadensis. Regarding this pull request specifically, Cyphal is the correct word to use.

File: src/drivers/cyphal/Cyphal.cpp [hunk 1]
```diff
@@ -305,101 +228,62 @@ void UavcanNode::Run()
...
-void UavcanNode::print_info()
+void CyphalNode::print_info()
 {
 	pthread_mutex_lock(&_node_mutex);
 
 	perf_print_counter(_cycle_perf);
 	perf_print_counter(_interval_perf);
 
-	O1HeapDiagnostics heap_diagnostics = o1heapGetDiagnostics(uavcan_allocator);
+	/*O1HeapDiagnostics heap_diagnostics = o1heapGetDiagnostics(uavcan_allocator);
```
Comment (JacobCrabill) [inline src/drivers/cyphal/Cyphal.cpp:248]: Why comment this out?

Comment (PetervdPerk-NXP) [reply src/drivers/cyphal/Cyphal.cpp:248]: Oops that went wrong migrating O1Heap to the CanardHandle class, I've created a new PR to fix this.

**Pull request merged**